reference/plugins.md

```diff
@@ -2,7 +2,170 @@
 layout: page
 title: Plugins
 ---
-We're working on building a way for you to extend Praxis. Please join the
-[development mailing
-list](https://groups.google.com/forum/#!forum/praxis-development) if you'd like
-to join that conversation.
+
+## Overview
+
+Plugins provide a means to cleanly add new functionality to Praxis, either by extending the core classes with additional features, or by registering callbacks to run during request handling.
+
+Presently, extensions to the following classes are supported by the plugin loader, along with the primary reason to extend them:
+
+ * `ResourceDefinition`, and `ActionDefinition`: additional DSL methods, e.g. `ActionDefinition#requires_authentication` directive that specifies that the action requires an authenticated user.
+ * `Request`: encapsulate access to the Rack environment hash. e.g., a `Request#current_user` method that returns a `User` instance for the current logged-in user.
+ * `Controller`: additional logic, and filters on actions. e.g., a `before :action` filter to verify the user's authentication for every request.
```

Comment:
I think the class / description / example is hard to follow. Maybe examples could be listed in a sub-list below each class / description. I wonder if it could help to word these in a more action-oriented way:
Add a `Request#current_user` method to access the currently logged-in user from any request object